### DISCUSSION TITLE
Graphics?

### DISCUSSION THREAD

**Discussion by SimpleSamples** [tui-cs/Terminal.Gui#1458] [General]
Does it use graphics? If not then it is not a GUI.

**Comment by tznind** [comment]
Please see [Graph View Scenario](https://github.com/migueldeicaza/gui.cs/blob/main/UICatalog/Scenarios/GraphViewExample.cs) in UICatalog for how to do graph

![image](https://user-images.githubusercontent.com/31306100/132085150-6c087f7b-7c44-4bd8-8db3-6d91ef459df9.png)

See also this discussion for pictures, all PR/feedback is welcome:
https://github.com/migueldeicaza/gui.cs/discussions/1265

**Reply by SimpleSamples** [reply]
That looks like text used to make a graph. It is a graph but it does not use graphics.

**Comment by BDisp** [comment]
This isn't a graph project but a multi-platform GUI Console solution, I think.

**Reply by SimpleSamples** [reply]
A GUI is a **Graphical** User Interface. The word graphical is intended to differentiate a graphical UI from a textual UI.

**Reply by BDisp** [reply]
Well I'm curious to see this repo as graphic :-)

**Reply by tznind** [reply]
The controls in this repo replicate those found in traditional GUI toolkits (buttons, windows, list/tree views etc).  I have linked a discussion above about possible future support for Sixel /  iTerm2 graphics.  So in the future icons and such could be supported.

Perhaps if you describe your needs that might help? or are you just disputing the appropriateness of the  name of the framework?

**Comment by SimpleSamples** [comment]
I was curious about textual user interfaces. Microsoft has developed an improved console system called pseudo consoles for Windows.  Windows consoles, both the classical type and the new pseudo consoles, do not support graphics. Windows consoles need textual UIs, not GUIs.

**Reply by BDisp** [reply]
The graphic term in a `GUI` solution means it is a graphical simulation of purely high-definition controls compounds of pixels, but using text only. Eventually we can use a unicode sequences that can simulate small images, such as emojis, but always based on lines and columns. The preference of many users is to have a lightweight application that works fast on a terminal, remote or not, with a more appealing interface, hence the `GUI` is called.

**Comment by tig** [comment]
This is a hilarious discussion. 

This entire discussion pivots on what one considers to be a pixel. 

Terminal.Gui's pixels are a character (glyph) cell. 

This means that compared to many other graphical systems, where the richness of each pixel is limited to just what color the pixel is (and sometimes brightness), Terminal.Gui is a FAR more capable graphical system, because every Terminal.Gui pixel can not only be whatever color you want, but also show one of many thousand glyphs.

**Comment by SimpleSamples** [comment]
I was looking for a text-only UI.

**Comment by tig** [comment]
But, what IS text?

-cek
Sent via mobile. Expect brevity, typos, and dangerous driving.
________________________________

**Comment by tznind** [comment]
The first step away from text is to free it of its rigid structure in straight consecutive lines:

![image](https://user-images.githubusercontent.com/31306100/132260438-f50f327c-843a-4623-9235-b6da84a5e2a4.png)
_From the novel House of Leaves_

The next step is to diversify the symbols used and add color:

![image](https://user-images.githubusercontent.com/31306100/132302473-fca9a566-3b78-46da-8134-9c3a007bc4fc.png)
_1337 Doom.  An ASCII rendering engine for the classic video game Doom_

Then you can embed graphics:

![shot-2021-09-06_22-13-37](https://user-images.githubusercontent.com/31306100/132260952-71f31c14-84f9-406a-bbf8-d8f288c32009.jpg)
_Console web browser w3m running in Linux_

At what point does it stop being a text UI and start being a graphics UI? To my mind there's no point and I don't think that really matters.  Whatever works for the user.
